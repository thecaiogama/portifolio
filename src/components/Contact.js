import React from 'react';

function Contact(props) {
  const { email } = props;
  return <contact>{email}</contact>;
}

export default Contact;
