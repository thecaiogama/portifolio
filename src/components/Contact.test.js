import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';
import Contact from './Contact';

describe('<Contact />', () => {
  const props = {
    email: 'emailteste@gmail.com'
  };

  exports[
    `test should render an email`
  ] = `<div id="contact">emailteste@gmail.com</div>`;

  it('should render an email', () => {
    const email = shallow(<Contact {...props} />);
    expect(email).toMatchSnapshot();
  });
});
