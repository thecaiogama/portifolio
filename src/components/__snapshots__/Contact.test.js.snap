// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render an email 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contact
    email="emailteste@gmail.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "emailteste@gmail.com",
      "id": "contact",
    },
    "ref": null,
    "rendered": "emailteste@gmail.com",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "emailteste@gmail.com",
        "id": "contact",
      },
      "ref": null,
      "rendered": "emailteste@gmail.com",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
